### ISSUE TITLE
HDD access issues with certain games.

### ISSUE THREAD

**Issue by jfmherokiller** [maharmstone/btrfs#718]
I don't know the true reason why this is occurring, but when playing games like idle colony, the game will have moments of lag.

This is tested using an NTFS partition on the same HDD, where the lag doesn't seem to occur. 

The lag seems to occur when the game seems to change music tracks and the game will have a semiregular solid few second long freeze before it can continue.

**Comment by Baconing** [comment]
I am having the same issue. Lots of stuttering and some (most) games are just straight up unplayable. 

Moving them over to my SSD works fine and causes no lag. Playing games on Linux on the HDD seems to work fine as well. Issues started cropping up recently, I've had this drive on BTRFS for a year or so now and I'd really only say it's been a problem for the past couple of months.

I am going to try booting into Linux and defragmenting the drive soon to see if that will help (since there's no support for it on WinBtrfs) but considering how games work fine on Linux I doubt it.

**Comment by Baconing** [comment]
Hey, I will say that [defragmenting the partition on Linux using the btrfs command](https://btrfs.readthedocs.io/en/latest/btrfs-filesystem.html#man-filesystem-cmd-defragment) helped a LOT. 

I'm not sure what games you're planning to run off an HDD but Portal and CS2 worked infinitely better. Not only am I getting less stutters but I am also getting much better loading times.

If you can, you should 100% try to defragment your partition on Linux. If you plan on booting into Linux often, you can also enable autodefrag in the filesystem options on Linux (e.g the fstab), but you should definitely try a full defrag for a couple of hours and see if it makes a difference.

I will say it's a bit strange that, for me personally, I didn't seem to have the stuttering issues on Linux and it only occured on Windows. My best guess is that the btrfs driver on linux does some magic that helps make random access on fragmented data more efficient.

tldr: defragment your drive

**Comment by jfmherokiller** [comment]
tbh I did that after posting this and it worked good.

**Comment by jfmherokiller** [comment]
I have encounteed another issue and its with pcem 13. 

For some reason the executable files remain open long after the process has closed. 

this is the error message 

![Image](https://github.com/user-attachments/assets/f5815cd3-47de-456e-adc7-7f0c0ba84f0a)

**Comment by ReisMiner** [comment]
I have the same issue in certain games but even on an m2 ssd. Helldivers 2 is an example where it stutters for 1s every 1min while elden ring works fine.

**Comment by Baconing** [comment]
> I have encounteed another issue and its with pcem 13.
> 
> For some reason the executable files remain open long after the process has closed.
> 
> this is the error message
> 
> ![Image](https://github.com/user-attachments/assets/f5815cd3-47de-456e-adc7-7f0c0ba84f0a)

I don't have this issue. Sometimes I do have an issue with the HDD being fully locked by processes and I need to reboot. (e.g. I try to open it and it says the folder is in use by another process and its unreadable)

**Comment by jfmherokiller** [comment]
well you are encountering a massive version of the issue I encountered. For me thankfully it was just a single process.

**Comment by Baconing** [comment]
Yeah, I also need to reboot my entire computer. I try using something like [File Locksmith](https://learn.microsoft.com/en-us/windows/powertoys/file-locksmith) to no avail. It says that explorer has the handle but restarting explorer it still holds.
